Swift/FleeceDecoder.swift

```diff
@@ -0,0 +1,531 @@
+            }
+        default:
+            throw CBLError.create(CBLError.decodingError, description: "Type mismatch: expected Array but found \(String(describing: decoder.fleeceValue))")
+        }
+    }
+    
+    func decode(_ type: Dictionary<String, any Decodable>.Type) throws -> Dictionary<String, any Decodable> {
+        switch decoder.fleeceValue {
+        case .dictionary(let dict):
+            if let dict = dict.toDictionary() as? Dictionary<String, any Decodable> {
+                return dict
+            } else {
+                fatalError("Failed to decode dictionary \(String(describing: dict.toDictionary())) as Decodable")
+            }
+        default:
+            throw CBLError.create(CBLError.decodingError, description: "Type mismatch: expected Dictionary but found \(String(describing: decoder.fleeceValue))")
+        }
+    }
+    
+    func decode<T>(_ type: T.Type) throws -> T where T: Decodable {
+        switch type {
+        case is Bool.Type:
+            return try decode(Bool.self) as! T
+        case is Int.Type:
+            return try decode(Int.self) as! T
+        case is Int8.Type:
+            return try decode(Int8.self) as! T
+        case is Int16.Type:
+            return try decode(Int16.self) as! T
+        case is Int32.Type:
+            return try decode(Int32.self) as! T
+        case is Int64.Type:
+            return try decode(Int64.self) as! T
+        case is UInt.Type:
+            return try decode(UInt.self) as! T
+        case is UInt8.Type:
+            return try decode(UInt8.self) as! T
+        case is UInt16.Type:
+            return try decode(UInt16.self) as! T
+        case is UInt32.Type:
+            return try decode(UInt32.self) as! T
+        case is UInt64.Type:
+            return try decode(UInt64.self) as! T
+        case is Float.Type:
+            return try decode(Float.self) as! T
+        case is Double.Type:
+            return try decode(Double.self) as! T
+        case is String.Type:
+            return try decode(String.self) as! T
+        case is Date.Type:
+            return try decode(Date.self) as! T
+        case is Blob.Type:
+            return try decode(Blob.self) as! T
+        case is Array<any Decodable>.Type:
+            return try decode(Array<any Decodable>.self) as! T
+        case is Dictionary<String, any Decodable>.Type:
+            return try decode(Dictionary<String, any Decodable>.self) as! T
+        default:
+            throw CBLError.create(CBLError.decodingError, description: "Type mismatch: expected \(T.self) but found \(String(describing: decoder.fleeceValue))")
+        }
+    }
+}
+
+enum FleeceValue {
+    case null
+    case bool(Bool)
+    case int(Int)
+    case uint(UInt)
+    case float(Float)
+    case double(Double)
+    case string(String)
+    case blob(Blob)
+    case array(ArrayObject)
+    case dictionary(DictionaryObject)
+    
+    private init?(_ value: Any) {
```

Comment:
I think I'm a bit disconnected here. Why FleeceValue only has Int and UInt type? If it means to target the largest value, should it be Int64 and UInt64?

Reply:
Hmmm that's a good point